src/stm32f411/clock.cpp

```diff
@@ -84,24 +85,25 @@ void configure_pll(clock_tree const& p_clock_tree)
     bit_modify(rcc->pllconfig)
       .insert(rcc_pllcnfg::input_division_factor, input_division_factor);
 
-    uint8_t output_division_factor = 0;
+    float output_division_factor = 0;
     /// 2 * (output_division_factor + 1) is to get the PLLP division factor
     /// Page 105 on the RM0383 User Manual
-    while (p_clock_tree.pll.output * (2 * (output_division_factor + 1)) <
+    while (p_clock_tree.pll.output * (2.0f * (output_division_factor + 1.0f)) <
            100.0_MHz) {
-      output_division_factor++;
+      output_division_factor += 1;
     }
 
     bit_modify(rcc->pllconfig)
-      .insert(rcc_pllcnfg::main_division_factor, output_division_factor);
+      .insert(rcc_pllcnfg::main_division_factor,
+              static_cast<std::uint8_t>(output_division_factor));
 
     /// 2 * (output_division_factor + 1) is to get the PLLN multiplicaiton
     /// factor Page 105 on the RM0383 User Manual
-    uint16_t target =
-      p_clock_tree.pll.output * (2 * (output_division_factor + 1)) / vco_in;
+    auto const target = p_clock_tree.pll.output *
+                        (2.0f * (output_division_factor + 1.0f)) / vco_in;
     bit_modify(rcc->pllconfig)
       .insert(rcc_pllcnfg::main_multiplication_factor,
-              static_cast<uint16_t>(target - 1));
+              static_cast<uint16_t>(target - 1.0f));
```

Comment:
Just curious, what's the idea behind switching all the variables to floats?

Reply:
This was to resolve clang-tidy warnings regarding narrowing the precision of an integer to a float. Even though the values are `u16` the operations transform the operations to `int` and integers may have 32-bits of precision where as `float`s have 23 bits of precision. So these changes were made to resolve these resolution issues. I tried to keep all of the logic the same. I also changed `1` to `1.0f` to ensure that the operations all stay as floats until the end.